card/mod/05_standard/set/right/account.rb

```diff
@@ -97,35 +77,34 @@ def token_error tcard, val
 end
 
 event :set_default_status, on: :create, before: :process_subcards do
-  default_status = ( Auth.needs_setup? ? 'active' : 'pending' )
+  default_status = Auth.needs_setup? ? 'active' : 'pending'
   add_subfield :status, content: default_status
 end
 
 def confirm_ok?
-  Card.new( type_id: Card.default_accounted_type_id ).ok? :create
+  Card.new(type_id: Card.default_accounted_type_id).ok? :create
 end
 
-event :generate_confirmation_token, :on=>:create, :before=>:process_subcards, :when=>proc{ |c| c.confirm_ok? } do
+event :generate_confirmation_token,
+      on: :create, before: :process_subcards,
+      when: proc { |c| c.confirm_ok? } do
   add_subfield :token, content: generate_token
 end
 
-event :reset_password, on: :update, before: :approve, when: proc{ |c| c.has_reset_token? } do
-  case ( result = authenticate_by_token @env_token )
-  when Integer
-    Auth.signin result
+event :reset_password, on: :update, before: :approve, when:
+    proc { |c| c.reset_password? } do
+  if validate_token! @env_token
+    token_card.used!
+    Auth.signin left_id
     success << edit_password_success_args
-    abort :success
-  when :token_expired
-    send_reset_password_token
-    success << {
-      id: '_self',
-      view: 'message',
-      message: "Sorry, this token has expired. Please check your email for a new password reset link."
-    }
-    abort :success
   else
-    abort :failure, "error resetting password: #{result}" # bad token or account
+    error_msg = errors.first.last
+    send_reset_password_token
+    msg = "Sorry, #{error_msg}. " \
+          'Please check your email for a new password reset link.'
+    success << { id: '_self', view: 'message', message: msg }
```

Comment:
this is a functional change.  Any time there is a token problem, we just send out a new email with a new token.  Simplifies the code, and is mostly what people want.